lib/Cake/View/Helper/FormHelper.php

```diff
@@ -2866,7 +2866,7 @@ protected function _selectOptions($elements = array(), $parents = array(), $show
 					}
 				}
 
-				if ($showParents || (!in_array($title, $parents))) {
+				if ($showParents || (!in_array($title, $parents, true))) {
```

Comment:
@dereuromark How about this? Although I'm not sure it covers the all cases what you indicated.
https://github.com/cakephp/cakephp/compare/2.x...kanonji:yet_another_fix?expand=1